Dynamic logging for connect (#3501)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc

.`logging`
Kafka Connect has its own configurable loggers:

* `log4j.rootLogger`
* `log4j.logger.org.reflections`

> Further loggers are added dependently on running Kafka Connect plugins.

Reviewer: ```suggestion
Further loggers are added depending on the Kafka Connect plugins running.
```

> To get complete list of connect loggers run from any Kafka broker pod:

Reviewer: ```suggestion
To get a complete list of Kafka Connect loggers running from any Kafka broker pod, use:
```

[source,curl,subs=attributes+]
----
curl -s http://<connect-cluster-name>-connect-api:8083/admin/loggers/
----

Kafka Connect uses the Apache `log4j` logger implementation.

> Use the `logging` property to configure loggers and logger levels.

Reviewer: Do we need to say what levels can be used?

Author: Mm, maybe yes. We do list logger levels for example in https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/modules/cruise-control/proc-deploying-cruise-control.adoc

  # ...
----

> All loggers which are available and are not configured have level set to `OFF`.

Reviewer: ```suggestion
Any available loggers that are not configured have their level set to `OFF`.
```

> Changes in Kafka Connect logging levels are applied dynamically.

Reviewer: ```suggestion
Changes to Kafka Connect logging levels are applied dynamically.
```

> If you use external logging and logging appenders are changed the rolling update is triggered.

Reviewer: ```suggestion
If you use external logging, a rolling update is triggered when logging appenders are changed.
```
